### ISSUE TITLE
can't install package

### ISSUE THREAD

**Issue by kyhwsk** [wbond/package_control#1011]
hi , I installed the sublime text 3(3083) , these days when i install the package, and i have set the  http_proxy the console shows error  
```
"http_proxy": "http://127.0.0.1:1080", //my shadow socks client
"https_proxy": "http://127.0.0.1:1080",
```
```
Exception in thread Thread-25:
Traceback (most recent call last):
  File "package_control.package_manager in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 371, in list_repositories
  File "package_control.providers.channel_provider in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 196, in get_repositories
  File "package_control.providers.channel_provider in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 87, in fetch
  File "package_control.download_manager in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 340, in fetch
  File "package_control.downloaders.urllib_downloader in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 224, in download
Package Control.package_control.downloaders.downloader_exception.DownloaderException: <unprintable DownloaderException object>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./threading.py", line 901, in _bootstrap_inner
  File "package_control.commands.install_package_command in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 46, in run
  File "package_control.package_installer in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 59, in make_package_list
  File "package_control.package_manager in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 453, in list_available_packages
  File "package_control.package_manager in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 413, in list_repositories
  File "package_control.console_write in /Users/baidu/Library/Application Support/Sublime Text 3/Installed Packages/Package Control.sublime-package", line 34, in console_write
TypeError: __str__ returned non-string (type NoneType)
```
but I dont know the reason,  how can i resolve it ?

**Comment by kyhwsk** [comment]
Besides, i have set the http_proxy

**Comment by wbond** [comment]
Unfortunately SOCKS proxies are not supported. You'll need an HTTPS proxy that supports `CONNECT`.

**Comment by wbond** [comment]
The exception is fixed by 3579f96e4f762a7145145f2cebe2b7b615867bc5
